src/threading/thread.cpp

```diff
@@ -196,6 +196,10 @@ void Thread::threadProc(Thread *thr)
 	thr->m_retval = thr->run();
 
 	thr->m_running = false;
+	// Unlock m_start_finished_mutex to prevent data race condition on Windows.
+	// On Windows with VS2017 build TerminateThread is called and this mutex is not
+	// released. We try to unlock it from caller thread and it's refused by system.
+	thr->m_start_finished_mutex.unlock();
```

Comment:
shouldn't this be inside an `#ifdef _WIN32` then?

Reply:
on android it's also undefined behaviour, we use pthread_kill, and it's better to do it properly